### ISSUE TITLE
Crashes due to race conditions when navigating

### ISSUE THREAD

**Issue by jgraham** [servo/servo#7677]
Currently there is a bunch of asynchronous state set up in the constellation and script task during load/paint that is implicitly assumed to be fully initialized by the time we come to navigate. This leads to race conditions that are relatively easy to lose by starting servo and immediately using webdriver to perform a navigation. For example #7675 covers the case of load then navigate calling `freeze` on a script task that doesn't yet have a root page. There are similar problems with that patch if you load/navigate/go back and try to return to a document that didn't fully load in the first instance.

**Comment by nox** [comment]
@asajeffrey Did the situation improve enough that we can close this issue?

**Comment by asajeffrey** [comment]
@nox: I doubt it, our constellation and content code still have lots of dubious synchronization assumptions, sigh. @jgraham?

**Comment by DemiMarie** [comment]
Fixed?

**Comment by jdm** [comment]
Yeah, this is non-specific and we've made many improvements around navigating and webdriver.
